_scripts/build-storybooks.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash -ex
+
+# TODO: only build if the current commit has changes to packages with storybooks - see modules-to-test.js for some hints?
+
+# Config from env vars with defaults
+SKIP_STORYBOOK_BUILD="${SKIP_STORYBOOK_BUILD:-false}"
+PROJECT_REPO="${PROJECT_REPO:-mozilla/fxa}"
+STORYBOOKS_REPO="${STORYBOOKS_REPO:-lmorchard/fxa-storybooks}"
+STORYBOOKS_URL="${STORYBOOKS_URL:-https://lmorchard.github.io/fxa-storybooks/}"
+STORYBOOKS_BRANCH="${STORYBOOKS_BRANCH:-gh-pages}"
+PUBLISH_ROOT="${PUBLISH_ROOT:-storybooks-publish}"
+MAX_STORYBOOK_AGE="${MAX_STORYBOOK_AGE:-90}"
+
+# Context for current build
+COMMIT_HASH=$(git rev-parse HEAD)
+
+# Build all the storybooks
+if [ "$SKIP_STORYBOOK_BUILD" == false ]; then
+    # fxa-payments-server relies on fxa-content-server .scss styles, which in turn
+    # rely on some modules in package.json
+    pushd ./packages/fxa-content-server
+    # npm ci
+    popd
+
+    STORYBOOKS=$(find packages -maxdepth 2 -type d -name '.storybook');
+    for STORYBOOK_PATH in $STORYBOOKS; do
+        pushd $(dirname $STORYBOOK_PATH);
+        rm -rf storybook-static;
+        # npm ci;
+        npm run build-storybook;
+        popd
+    done;
+fi
+
+# Get a fresh checkout of the repo to which we'll publish
+rm -rf $PUBLISH_ROOT
+git clone --branch $STORYBOOKS_BRANCH git@github.com:$STORYBOOKS_REPO.git $PUBLISH_ROOT
+mkdir -p $PUBLISH_ROOT/{commits,pulls}
+
+# Copy all the storybooks into the publish directory for commit.
+COMMIT_PATH="$PUBLISH_ROOT/commits/$COMMIT_HASH"
+rm -rf $COMMIT_PATH
+mkdir -p $COMMIT_PATH
+for STORYBOOK_DIR in $(find -type d -name 'storybook-static'); do
+    PACKAGE_NAME=$(basename $(dirname $STORYBOOK_DIR));
+    BUILD_PATH="$COMMIT_PATH/$PACKAGE_NAME";
+    cp -r $STORYBOOK_DIR $BUILD_PATH
+done
+
+# HACK: fixup some generated CSS paths to static media that break since the
+# storybooks are no longer at the root of the site. Would be better to
+# figure out how to reconfigure storybook to do this, but I got here faster
+for CSS_FN in $(find $COMMIT_PATH -type f -name 'main*.css'); do
+    sed --in-place 's:url(static/:url(../../static/:g' $CSS_FN
```

Comment:
`sed --in-place` errors for me in OS X, as discussed in Slack I thought `sed -i "" -e` in OSX was the solution but it doesn't appear to fix it so eh, maybe there's an alternative text replace.

Reply:
Yeah, this one might be hard to find a fix for. The `sed` we'll use in CI will be the GNU version - so if the BSD version on OS X disagrees, we might not be able to easily run this as-is on OS X without maybe a check for `darwin` in `uname -a` or something tangly like that